### PR TITLE
refactor(console): promote and re-use get-started component in both applications and connectors

### PR DESCRIPTION
**Pull request by demonzoo**
## Summary

* Refactor application get-started guide component by extracting sub-components and custom hooks.
* Promote <GetStarted> component to higher level, in order to make it re-usable in both applications and connectors

## Testing

- [x] Tested in browser, and the application get-started page logics remained the same.

image

### CODE REVIEW

**Review by wangsijie — APPROVED** [review]

File: packages/console/src/pages/Applications/components/LibrarySelector/index.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import RadioGroup, { Radio } from '@/components/RadioGroup';
+import { SupportedJavascriptLibraries } from '@/types/applications';
+
+import * as styles from './index.module.scss';
+
+type Props = {
+  className?: string;
+  libraryName?: SupportedJavascriptLibraries;
+  onChange?: (value: string) => void;
+  onToggle?: () => void;
```
Comment (simeng-li) [inline packages/console/src/pages/Applications/components/LibrarySelector/index.tsx:18]: nit: do we need to pass the target library name here

File: packages/console/src/pages/GetStarted/index.tsx [hunk 2]
```diff
@@ -1,7 +1,126 @@
...
+  onClose,
+  onComplete,
+  onToggleSteps,
+}: Props) => {
+  const [subtype, setSubtype] = useState(defaultSubtype);
+  const [activeStepIndex, setActiveStepIndex] = useState<number>(-1);
+  const steps = useGetStartedSteps(type, subtype) ?? [];
+
+  const stepReferences = useMemo(
+    () => Array.from({ length: steps.length }).map(() => createRef<HTMLDivElement>()),
```
Comment (simeng-li) [inline packages/console/src/pages/GetStarted/index.tsx:58]: nit: Array.from().fill() ?

File: packages/console/src/pages/GetStarted/index.tsx [hunk 3]
```diff
@@ -1,7 +1,126 @@
...
+              onNext={() => {
+                setActiveStepIndex(index + 1);
+              }}
+              onToggle={() => {
+                setActiveStepIndex(index);
+                onToggleSteps?.();
+              }}
+            />
+          );
+        })}
```
Comment (simeng-li) [inline packages/console/src/pages/GetStarted/index.tsx:120]: extract to a use memo?

Comment (demonzoo) [reply packages/console/src/pages/GetStarted/index.tsx:120]: We just talked about it and we don't think a useMemo would help in this case. Resolved.

File: packages/console/src/pages/Applications/components/LibrarySelector/index.tsx (hunk 1 above)
Comment (demonzoo) [reply packages/console/src/pages/Applications/components/LibrarySelector/index.tsx:18]: Yes, the library name is the default value of this selector component.

File: packages/console/src/pages/GetStarted/index.tsx (hunk 2 above)
Comment (demonzoo) [reply packages/console/src/pages/GetStarted/index.tsx]: Updated

**Pull request merged**